mern-ecommerce/controllers/braintree.js

```diff
@@ -21,8 +21,8 @@ exports.generateToken = (req, res) => {
 };
 
 exports.processPayment = (req, res) => {
-  let nonceFromTheClient = req.body.paymentMethodNonce;
-  let amountFromTheClient = req.body.amount;
+  let nonceFromTheClient = request.body.paymentMethodNonce;
+  let amountFromTheClient = request.body.amount;
```

Comment:
The variable `request` is not defined in this scope. It should be `req` instead of `request`.